mvrx/src/main/kotlin/com/airbnb/mvrx/FlowExtensions.kt

```diff
@@ -0,0 +1,73 @@
+package com.airbnb.mvrx
+
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.dropWhile
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.plus
+import kotlinx.coroutines.yield
+import java.util.concurrent.ConcurrentHashMap
+
+internal fun <T : Any?> Flow<T>.collectLatest(
+    lifecycleOwner: LifecycleOwner,
+    lastDeliveredStates: ConcurrentHashMap<String, Any?>,
+    activeSubscriptions: MutableSet<String>,
+    deliveryMode: DeliveryMode,
+    action: suspend (T) -> Unit
+): Job {
+    val flow = when {
+        MavericksTestOverrides.FORCE_DISABLE_LIFECYCLE_AWARE_OBSERVER -> this
+        deliveryMode is UniqueOnly -> {
+            this.assertOneActiveSubscription(lifecycleOwner, activeSubscriptions, deliveryMode.subscriptionId)
+                .dropWhile { it == lastDeliveredStates[deliveryMode.subscriptionId] }
```

Comment:
if `lastDeliveredStates[deliveryMode.subscriptionId]  == null` you can't distinguish if it's missing or it's null, would be better to do
```suggestion
                .dropWhile { lastDeliveredStates.contains(deliveryMode.subscriptionId) && it == lastDeliveredStates[deliveryMode.subscriptionId] }
```

Reply:
yeah, this was an existing bug, but would be good to fix

Reply:
previously it did work because `lastDeliveredStates` contained only non-null objects (and all nullable fields were wrapped into MavericksTuple1), so null was the clear indicator the key is missing in the map

Reply:
hmm yea you are right. previously every that was wrapped in a mavericks tuple would still work but if not it could behave weird. the suggested change would also catch this as well. updating